doc/source/reference/c-api.types-and-structures.rst

```diff
@@ -355,13 +355,14 @@ PyArrayDescr_Type
 
 .. c:member:: PyObject *PyArray_Descr.fields
 
-    If this is non-NULL, then this data-type-descriptor has fields
+    For built-in data types, this is set to ``Py_None``. Otherwise,
```

Comment:
Yes, for non-builtin dtypes without any fields. I guess the wording needs improving?